### DISCUSSION TITLE
older commits without .lfsconfig file

### DISCUSSION THREAD

**Discussion by sam-cha-tfs** [git-lfs/git-lfs#5460] [Q&A] [answered]
What happens if you're on a branch with a .lfsconfig file and switch to an older commit in the branch that doesn't contain the .fsconfig file? Will GIT know where to pull LFS objects contained in the repo?

**Comment by chrisd8088** [comment]
I think the brief answer is likely no, Git LFS won't know where to pull objects from in that case.

But it does depend on whether you have another configuration file like `.git/config` or `~/.gitconfig` which might supply the `lfs.url` value regardless of whether `.lfsconfig` exists.

It also depends on whether you have any Git LFS objects in the branch prior to the addition of the `.lfsconfig` file.  If there are no Git LFS objects, then it probably doesn't matter that you don't have an `lfs.url` configuration at that point in your Git history.

**Reply by sam-cha-tfs** [reply]
So basically it seems like what you're saying is we can't checkout older commits that contain LFS objects and not the config files. 
Wouldn't this mean that a repo using ` git lfs migrate import --everything above 100MB` to migrate to GitHub can only checkout at the point of the migration?

**Reply by chrisd8088** [reply, marked as answer]
It's probably worth reviewing the `git-lfs-config(5)` manual [page](https://github.com/git-lfs/git-lfs/blob/1820b646e05c24497652b3f9eaf91500dc1ee988/docs/man/git-lfs-config.adoc), which adds a few details, particularly about the `lfs.url` configuration value:
> The url used to call the Git LFS remote API. Default blank (derive from clone URL).

So in general one doesn't need to specify that value explicitly at all, you can freely checkout any commit and Git LFS should be able to retrieve objects (or push them) from the default API.

I'm assuming from your question, though, that you want to override the default with a specific non-default value, which is common when using `.lfsconfig` because when you have a non-default Git LFS remote API, it's valuable to include that configuration in the repository contents so that it's shared by all developers.  That's a key use case for `.lfsconfig`.

Under those circumstances, I think you'll want to ensure that the `.lfsconfig` file exists in your Git history for any commit containing a Git LFS object.

If you're using `git lfs migrate import` to rewrite your Git history so as to use Git LFS for certain objects, then you can add the `.lfsconfig` file by rebasing so it's included in the earliest commits with Git LFS objects.

Again, though, this all assumes you need `.lfsconfig` because you have some Git LFS configuration you want to share with multiple developers and, in particular, a non-default Git LFS API endpoint is one of those configuration values.

**Reply by sam-cha-tfs** [reply]
Thank you for your answer. Yes you are correct I plan to use JFROG to store the LFS objects therefore the .lfsconfig is needed.
